File: src/domain/usecases/ResetDemoUseCase.ts
import { ILearnerRepository } from '../repositories/ILearnerRepository';
import { LearnerProfile } from '../entities/LearnerProfile';

export class ResetDemoUseCase {
  constructor(private learnerRepo: ILearnerRepository) {}

  async execute(): Promise<LearnerProfile> {
    return this.learnerRepo.resetAll();
  }
}

File: mobile/lib/domain/usecases/reset_demo_use_case.dart
import '../repositories/i_learner_repository.dart';
import '../entities/learner_profile.dart';

class ResetDemoUseCase {
  final ILearnerRepository learnerRepository;

  const ResetDemoUseCase(this.learnerRepository);

  Future<LearnerProfile> execute() async {
    return learnerRepository.resetAll();
  }
}
